### DISCUSSION TITLE
How can I push AV1 in RTSP?

### DISCUSSION THREAD

**Discussion by bluesky2319** [bluenviron/mediamtx#2078] [Questions] [answered]
I have a question about pushing AV1 (by hardware encoding of Nvidia) by MediaMTX and FFmpeg. Based on the description, MediaMTX supports AV1 in RTSP. I would like to know how to test and what version of FFmpeg is required. I would appreciate any sample FFmpeg command for it. 
Thank you.

**Comment by aler9** [comment, marked as answer]
FFmpeg doesn't yet support publishing SRT with RTSP. Contact FFmpeg authors in order to request the feature.

**Reply by bluesky2319** [reply]
Thanks for your reply. What about GStreamer?

**Reply by aler9** [reply]
<s>The same applies for GStreamer. The server fully supports receiving AV1 with RTSP but major players have not implemented this feature yet. There's nothing we can do here about it.</s>

**Reply by bluesky2319** [reply]
Thanks again. Then, are there any other protocols we can use for publishing AV1 by FFmpeg, like DASH or enhanced RTMP?

**Reply by aler9** [reply]
My last comment was inaccurate: GStreamer >= 1.22 supports publishing AV1 with RTSP. Rust plugins are also needed.

I'm not aware of any publishing method supported by FFmpeg.

Anyway, i think the issue is gonna solve by itself since in the next years support for AV1 will be surely improved in all streaming software.

**Reply by bluesky2319** [reply]
That's great. Then we can use GStreamer with MediaMTX for publishing AV1 with RTSP. Thank you for your help.

[a comment by an automated account is not shown]
